hphp/util/light-process.cpp

```diff
@@ -436,6 +436,23 @@ void LightProcess::SigChldHandler(int sig, siginfo_t* info, void* ctx) {
   }
 }
 
+void LightProcess::AttachHandler(sighandler_t sighandler)
+{
+  if (sighandler) {
+    always_assert_flog(sighandler == (sighandler_t)&SigChldHandler,
```

Comment:
I see what you mean, but the comparison has to be done within the LightProcess class since it is the only class that know explicitly about its SigChldHandler. So I added this parameter which can be null or a sighandler_t. If it is a sighandler_t (not null), then the assert is there to ensure that we're only attaching SigChldHandler again if the saved handler was the same. If it is null, then we're attaching for the first time in LightProcess which won't check.

I'm happy to try something a different approach if you have any ideas.